### PR TITLE
Resolves Issue #7

### LINKED ISSUE #7: `-o` flag should also allow for path to location for file writing

### PR DESCRIPTION
**Pull request by gould-arthur**
Closes #7

### CODE REVIEW

File: pepmatch/matcher.py [hunk 1]
```diff
@@ -838,21 +836,31 @@ def filter_fragments(group: pd.DataFrame) -> pd.DataFrame:
...
-    elif self.output_format == 'tsv':
-      return df.to_csv(f'{path}.tsv', sep='\t', index=False)
-    elif self.output_format == 'xlsx':
-      return df.to_excel(f'{path}.xlsx', index=False)
+    # appends '.' + filetype if the name does not already contain it
+    path = self.output_name.__str__()
+    if path.split('.')[-1] != self.output_format:
+      path += f".{self.output_format}"
+
+    output = self._get_function_pointer(df)
```
Comment (danielmarrama) [inline pepmatch/matcher.py:859]: can we call `_get_function_pointer` `_get_output_function` instead?

File: pepmatch/tests/test_ouptput_files.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    _creation_calls(match, relative_location, simple_dataframe)
+
+    assert relative_location.exists()
+    remove(relative_location.__str__())
+
+
+def test_absolute_creation(match, simple_dataframe):
+    """Tests creation given an absolute path"""
+    absolute_location = Path("absolute.csv").absolute()
```
Comment (danielmarrama) [inline pepmatch/tests/test_ouptput_files.py:58]: I think you'll need to actually create a full path like: `/full/path/to/file.csv`

**Review by danielmarrama — COMMENTED** [review]
Overally, very good attempt. Some minor changes and then I'll merge.

One other thing I'd like: 2 spaces instead of 4 for indentation 😃

**Review by danielmarrama — CHANGES_REQUESTED** [review]
Comment for requesting changes

File: pepmatch/matcher.py (hunk 1 above)
Comment (gould-arthur) [reply pepmatch/matcher.py:859]: You bet

File: pepmatch/matcher.py [hunk 3]
```diff
@@ -838,21 +836,31 @@ def filter_fragments(group: pd.DataFrame) -> pd.DataFrame:
 
     return df
 
+  def _get_function_pointer(self, df: pd.DataFrame):
+    """Returns the df conversion function, given the required output format
+    
+    Args:
+      df: the dataframe of the matches."""
+    return [df.to_csv, df.to_csv, df.to_excel, df.to_json, df.to_html][
```
[the remark this answers is not in the document]
Comment (gould-arthur) [reply pepmatch/matcher.py:844]: I believe that df.to_tsv() does not exist, It is df.to_csv, with sep='\t'

[the remark this answers is not in the document]
Comment (danielmarrama) [reply pepmatch/matcher.py:844]: Yeah, I'm stupid. I've used this so many times and forgot that to_csv is used for .tsv so nvm this comment.

**Comment by gould-arthur** [comment]
Refactored the long lines near ln 800 in matcher.py
Only waiting for action approval, review on factored lines, and an answer to the question on absolute path.

File: pepmatch/matcher.py [hunk 4]
```diff
@@ -794,23 +795,19 @@ def filter_fragments(group: pd.DataFrame) -> pd.DataFrame:
...
-      idx = (df.groupby(
-        ['Query Sequence']
-      )['Protein Existence Level'].transform('min') == df['Protein Existence Level']) | df['Protein Existence Level'].isna()
-      df = df[idx]
+      # take matches that are in the gene priority proteome
+      df = df[self._take_matches(df, "Gene Priority", "max")]
+
+      # and with the best protein existence leve
+      idx = self._take_matches(df, "Protein Existence Level")
+      df = df[self._take_matches(df, "Protein Existence Level")]
```
Comment (danielmarrama) [inline pepmatch/matcher.py:810]: Just delete the first line and then I'll merge

**Review by danielmarrama — CHANGES_REQUESTED** [review]
One last change

**Comment by danielmarrama** [comment]
Very nice, Arthur. Thank you!!!!!!

**Pull request merged**